plugins/web-test-runner-plugin/plugin.js

```diff
@@ -0,0 +1,60 @@
+const {isTestFilePath} = require('@web/test-runner');
+const snowpack = require('snowpack');
+const fs = require('fs');
+const path = require('path');
+const url = require('url');
+const cwd = process.cwd();
+
+module.exports = function () {
+  if (process.env.NODE_ENV !== 'test') {
+    throw new Error(`@snowpack/web-test-runner-plugin: NODE_ENV must === "test" to build files correctly.
+To Resolve:
+  1. Set "process.env.NODE_ENV = 'test';" at the top of your web-test-runner.config.js file (before all imports).
+  2. Prefix your web-test-runner CLI command: "NODE_ENV=test web-test-runner ...".
+`);
+  }
+  const pkgManifest = require(path.join(cwd, 'package.json'));
+  const config = snowpack.unstable__loadAndValidateConfig(
+    {devOptions: {hmr: false, open: 'none', output: 'stream'}},
+    pkgManifest,
+  );
+  let loadByUrl, shutdownServer;
+
+  return {
+    name: 'snowpack-plugin',
+    async serverStart({fileWatcher}) {
+      fileWatcher.add(Object.keys(config.mount));
+      const server = await snowpack.unstable__startServer({
+        cwd,
+        config,
+        lockfile: null,
+        pkgManifest,
+      });
+      loadByUrl = server.loadByUrl;
+      shutdownServer = server.shutdown;
+    },
+    async serverStop() {
+      return shutdownServer();
+    },
+    async serve({request}) {
+      if (request.url.startsWith('/__web-dev-server')) {
+        return;
+      }
+      const reqPath = request.url.replace(/\?.*$/, '');
+      const result = await loadByUrl(reqPath, {isSSR: false});
+      return result;
+    },
+    transformImport({source}) {
+      if (!isTestFilePath(source) || source.startsWith('/__web-dev-server')) {
+        return;
+      }
+      const reqPath = source.replace(/\?.*$/, '');
```

Comment:
This is quite a hot path in the code, I wonder if `source.split('?')[0]` would be faster, or at least placing the regexp in a constant would avoid recreating it each run.

Reply:
Did some quick perf / code golf: 

https://perf.link/#eyJpZCI6ImhwcTl3Znh1eWV5IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IHNvdXJjZSA9ICcvd2ViX21vZHVsZXMvY29tbW9uL19jb21tb25qc0hlbHBlcnMtOGMxOWRlYzguanMnO1xuY29uc3QgU1RSSVBfUVNfUkVHRVggPSAvXFw%2FLiokLztcbiIsInRlc3RzIjpbeyJuYW1lIjoiRmluZCBpdGVtIDEwMCIsImNvZGUiOiJzb3VyY2Uuc3Vic3RyaW5nKDAsIHNvdXJjZS5pbmRleE9mKCc%2FJykgPT09IC0xID8gdW5kZWZpbmVkIDogIHNvdXJjZS5pbmRleE9mKCc%2FJykpIiwicnVucyI6WzEyOTcwMDAsMTk3MDAwLDI3NDgwMDAsMjEyNzAwMCw4NzYwMDAsMTc2OTAwMCwyMTkxMDAwLDExMjEwMDAsNzk5MDAwLDM2MTAwMDAsMTMzOTAwMCwzMzY3MDAwLDE3NzgwMDAsMTUwNjAwMCw2MjAwMCwyNzYzMDAwLDEyMzkwMDAsNTk1MDAwLDM1MzEwMDAsMzAxNTAwMCwxNDIxMDAwLDI1NzkwMDAsMTA3MDAwLDI1MTgwMDAsMTE4NzAwMCwyNTY1MDAwLDE3MzUwMDAsMjEzNTAwMCw1MDAwMCwxMDkwMDAwLDE2NTMwMDAsODg1MDAwLDE4MjQwMDAsMjk4ODAwMCwzNzk2MDAwLDI2MDAwLDQyMjAwMCwyOTYyMDAwLDMzODYwMDAsNjM5MDAwLDE4MDcwMDAsMzU5MjAwMCwyMzAwMCwxNjA1MDAwLDQ5NTAwMCwzMjg0MDAwLDIyMTkwMDAsMjY0OTAwMCwzMTY4MDAwLDc2NDAwMCwyNTgwMDAsMzAwMCwzOTI0MDAwLDMxMjUwMDAsMzYyNzAwMCw0MTAwMDAsMTkyNTAwMCw1NDEwMDAsNDkwMDAwLDI2NTAwMDAsMTYxOTAwMCwyMjc5MDAwLDE3NjEwMDAsMzAxMDAwLDEyMDAwLDE4OTkwMDAsMzkwNjAwMCwzMTU2MDAwLDI0NjAwMDAsOTAwMDAwLDM3NDUwMDAsODE2MDAwLDczODAwMCw4MjcwMDAsMzMwNjAwMCwyNDcxMDAwLDQxNTAwMCwxNzAyMDAwLDEzMDAwLDI0MDcwMDAsMzk1NTAwMCw5OTQwMDAsMjEzNDAwMCwxMDI2MDAwLDM2NTUwMDAsMzIxMDAwLDI2NDMwMDAsMzM1MTAwMCw4NjQwMDAsMzkzOTAwMCw4NDUwMDAsMTQ3MzAwMCwxNDA5MDAwLDI3OTQwMDAsMTE4NzAwMCw2NDAwMDAsMzUzMjAwMCw2ODIwMDAsMzI5MTAwMCw4MzAwMF0sIm9wcyI6MH0seyJuYW1lIjoiRmluZCBpdGVtIDIwMCIsImNvZGUiOiJjb25zdCBxdWVyeVN0cmluZ1N0YXJ0ID0gc291cmNlLmluZGV4T2YoJz8nKTtcbnNvdXJjZS5zdWJzdHJpbmcoMCwgcXVlcnlTdHJpbmdTdGFydCA9PT0gLTEgPyB1bmRlZmluZWQgOiBxdWVyeVN0cmluZ1N0YXJ0KSIsInJ1bnMiOls1ODUwMDAsMzI5MzAwMCwyNTkyMDAwLDE4NDIwMDAsNTAwMCwxNTk4MDAwLDI0NzcwMDAsOTk5MDAwLDEwNTYwMDAsMzEyNTAwMCwxMTI3MDAwLDMxNDUwMDAsMTc5NjAwMCwzODY5MDAwLDYyMDAwLDI4ODkwMDAsMTA4MzAwMCw3NTcwMDAsMzUyNzAwMCwyNjM0MDAwLDM3MDMwMDAsMjQ5MDAwMCw5NjQwMDAsMjM1OTAwMCwyOTI5MDAwLDUyNDAwMCwzMjYwMDAwLDIyMjcwMDAsMjg1MjAwMCwxMzA5MDAwLDE0MTMwMDAsMTE3MTAwMCwxOTI5MDAwLDE2ODIwMDAsMTYyNjAwMCwyODAwMCw0MzUwMDAsNjc3MDAwLDM2MjgwMDAsMzAyMDAwLDE5MDAwMDAsMzc4MjAwMCwyODAwMCwzMTgyMDAwLDE4ODYwMDAsNTg1MDAwLDIwNTIwMDAsMjEyMTAwMCwzMzk0MDAwLDUxNjAwMCwyNzczMDAwLDM4MjcwMDAsMzg5MzAwMCwzMDE1MDAwLDMxNDUwMDAsNDEwMDAwLDQwODAwMCwxODIwMDAwLDMyOTMwMDAsMjA3MTAwMCwxMTYyMDAwLDEwMzgwMDAsMTY5NjAwMCwyODY3MDAwLDMzNTAwMDAsMTA0NjAwMCwzODc2MDAwLDMxNDkwMDAsNTkyMDAwLDIxMjEwMDAsMzc5NTAwMCw4MjAwMDAsODE3MDAwLDYxNDAwMCwxOTMxMDAwLDgzMDAwMCwzMzEwMDAsMTYwMTAwMCwzMjQ0MDAwLDI3MzMwMDAsMjM1MzAwMCwzMTQxMDAwLDIwMjkwMDAsMTE4NDAwMCw3MzAwMCwzOTQwMDAsMjI3MDAwMCwxMDk3MDAwLDU4MDAwMCw0OTYwMDAsNjM5MDAwLDI4NTUwMDAsMTQ4NTAwMCwyNDkxMDAwLDE0NzMwMDAsMTAwNzAwMCwyODY1MDAwLDg0MTAwMCwxMjAwMDAsMjEzNDAwMF0sIm9wcyI6MH0seyJuYW1lIjoiRmluZCBpdGVtIDQwMCIsImNvZGUiOiJzb3VyY2UucmVwbGFjZShTVFJJUF9RU19SRUdFWCwgJycpOyIsInJ1bnMiOlsxMDAwLDI2NTgwMDAsMjE0MjAwMCw4MTIwMDAsMTAwMCwxMTA4MDAwLDE3MzcwMDAsMjkyMzAwMCw0NTIwMDAsMjU2MjAwMCwxMDE3MDAwLDIwMDAsNjMwMDAwLDMyMDAwMDAsMjQzOTAwMCw1MjQwMDAsMTAwMCwxMjAwMCwxMDAwLDI2MDEwMDAsMzA1NzAwMCwxNDk3MDAwLDc1MDAwMCwxNjY4MDAwLDIyODUwMDAsMjAwMCwxNjIwMDAwLDE0MzAwMDAsMjAwMCw4MzQwMDAsMjgzNTAwMCw3NTEwMDAsODA2MDAwLDg5OTAwMCwyMDAwLDIwMDAsNjczMDAwLDE4NjEwMDAsMTAwMCwyNjk1MDAwLDEyMDcwMDAsMjUyMzAwMCwyOTEyMDAwLDI1MTIwMDAsOTUxMDAwLDIxMDAwMDAsMTQ0ODAwMCwyMDAwLDI1MTYwMDAsMjQ1MDAwLDcyMzAwMCwxMzY0MDAwLDI2NzgwMDAsMjY1MDAwMCwyMDAwLDI4MzMwMDAsNTAwMCwyMTMwMDAsMjU2MjAwMCwxNzc3MDAwLDE0NjAwMCw3ODUwMDAsMTI0MTAwMCwyODQyMDAwLDE2MjAwMDAsMjY1MTAwMCwxMDAwLDIzNjEwMDAsMzAxMDAwLDE2MTIwMDAsMjAwMCwyMDAwLDU4NjAwMCwzNjIwMDAsOTA4MDAwLDM2NjAwMCwxNzY0MDAwLDE4MTgwMDAsMjEzMjAwMCwyNTgyMDAwLDIxNTgwMDAsMjUxMjAwMCwxMzMzMDAwLDc5NzAwMCwzMDM3MDAwLDIxNDAwMDAsOTU1MDAwLDIwMDAsMTcwMDAsMjAwMCw1MzAwMDAsMTg3NjAwMCwyMDAwLDE4ODQwMDAsMjA0MTAwMCwyNTgwMDAsMTk3NDAwMCwyMDAwLDIwMDAsNjM2MDAwXSwib3BzIjowfV0sInVwZGF0ZWQiOiIyMDIwLTEwLTExVDIwOjQ0OjA5LjgwMVoifQ%3D%3D